### ISSUE TITLE
feat: Allow using native window decorations/titlebar

### ISSUE THREAD

**Issue by ethanc8** [janhq/jan#3093]
### #

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

In versions before v0.5.x, Jan used the system's native window decorations (titlebar, window borders, shadow behind windows, etc). However, in v0.5.1, Jan draws its own titlebar and does not implement things like shadows behind the window or the platform-native right-click menu, at least on Linux with the KDE desktop environment. This makes Jan feel unnatural and annoying to use, as the lack of shadows behind the window cause the window to blend in to other windows, and it generally does not feel like a normal window.

### Describe the solution

As a bare minimum, an option should be added to the appearance settings to allow the native window decorations to be used. If possible, we should replace Jan's custom titlebar with the Electron menubar (File, Edit, Help...) when native window decorations are being used. Another thing we could look into is improving the behavior of the nonnative window decorations, such as implementing drop shadows and right-click on the titlebar. This is something that all Electron apps suffer from, while Gtk apps don't suffer as much because the Gtk theme typically implements behavior close to native for Gtk apps that draw their own window decorations.

### Teachability, documentation, adoption, migration strategy

On some platforms, the titlebar and window borders of the Jan main window may feel out of place or odd. If you are bothered by this, you can go to Settings > Appearance > Window Decorations, select "Native Window Decorations", and reopen the app.

### What is the motivation / use case for changing the behavior?

This behavior is a regression from v0.4.x. It also makes the menubar harder to access.

**Comment by philippgitpush** [comment]
+1, not having shadows behind its window makes resizing the window difficult when several windows are open that use the same color theme

![image](https://github.com/user-attachments/assets/0b934bf3-f773-4054-88e1-adbb7abe2c31)

**Comment by muhammad-halimov** [comment]
+1 I would like to see gnome native libadwaita/adw-gtk3 title bars support

**Comment by dan-homebrew** [comment]
@imtuyethan Can I check my understanding that this is resolved in #3324? 

I would prefer for us to close issues with adequate communcation to users, e.g. this is fixed in 0.5.4.

**Comment by urmauur** [comment]
Hi guys @ethanc8 @philippgitpush @muhammad-halimov we already fixed this issue for the next release but unfortunately after discuss with our designer we still kept our topbar instead of using electron native topbar

### Resolved issue
- snap window, u can user ctrl + left and right  also frag jan window into top to trigger snap grid windows feature
- shadow and border
- menubar with a hamburger icon 

![2024-09-12 14 14 23](https://github.com/user-attachments/assets/b000480d-7fa0-4d03-be07-6b57b85e5c0e)
![2024-09-12 14 14 14](https://github.com/user-attachments/assets/4955b37b-9121-45aa-91e6-6ad86fde985d)
![2024-09-12 14 14 05](https://github.com/user-attachments/assets/9410164b-bc39-4d95-9331-fb0dc11f009a)

cc @dan-homebrew @0xSage @imtuyethan 

please fell free give another feedback after next release

**Comment by ethanc8** [comment]
On Linux, problems almost always arise with nonnative topbars, especially for people using less popular themes or window managers. We still need a setting to disable this.

**Comment by imtuyethan** [comment]
> On Linux, problems almost always arise with nonnative topbars, especially for people using less popular themes or window managers. We still need a setting to disable this.

@ethanc8 can you elaborate more on this? Possibly with recordings of how you use other apps with a native top bar. 
As for Jan:
- We keep all actions in a Hamburger icon as for our philosophy of keeping things as much simple as possible. 
- We have resolved all of the problems like snapping window, etc...

Would be great to know more about the "problems" you mentioned, much easier for us.

**Comment by dan-homebrew** [comment]
@imtuyethan I think @ethanc8 is right - there's likely a lot of edge cases in Linux we are not dealing with. 

- The changes @urmauur made will solve for Windows, but not Linux
- There are likely a million Linux distros where this would break on (e.g. Mint, etc)

@urmauur: Is it possible to make this a setting that user can turn on/off? I think we should let users make the decision. 

> > On Linux, problems almost always arise with nonnative topbars, especially for people using less popular themes or window managers. We still need a setting to disable this.
> 
> @ethanc8 can you elaborate more on this? Possibly with recordings of how you use other apps with a native top bar.
> 
> As for Jan:
> 
> * We keep all actions in a Hamburger icon as for our philosophy of keeping things as much simple as possible.
> * We have resolved all of the problems like snapping window, etc...
> 
> Would be great to know more about the "problems" you mentioned, much easier for us. I can open another issue to follow up if the problems are solid. Thanks in advanced!

**Comment by imtuyethan** [comment]
@dan-homebrew @urmauur I created another ticket for this: https://github.com/janhq/jan/issues/3666

Thanks @ethanc8!

**Comment by dandv** [comment]
Should this sort of fancy design be the focus of an LLM chat tool?

Is it worth the time spent and the [annoyances](https://github.com/janhq/jan/issues/4570) it creates?

Has there been user demand for removing the OS window decorations?

**Comment by espdev** [comment]
In the latest version (v0.7.7), I can't even move the window properly with the mouse. There's only a tiny strip in the header that you can grab onto.

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/b0ce504f-9c01-4ff2-ab71-636d54046f20" />

I don't really understand the point of a design that is inconvenient to use.
Also, the window never remembered and does not remember its position and state after closing.

**Comment by hwertz** [comment]
I just started using Jan (from LM Studio); even though both are using llama-server (out of the box at least), Jan seems to have noticeably better performance for me (running models that are pushing my memory limits a bit).  Kudos to the devs!

Anyway, I'd prefer native window decoration too.  I have my desktop set to "middle click on title bar to put window below all other windows", except Steam and now Jan (Firefox defaults to a non-native title bar but I turned it off on there.)  I'm not using an exotic window manager, KDE with pretty much stock theme here, so it does look and work nicely already; this would just be a minor quality of life improvement in my case.  Pretty minor, but it's a use case where the title bar makes a small difference.

Thanks!
--Henry
